grapher/controls/Controls.tsx

```diff
@@ -390,9 +390,21 @@ export class FooterControls extends React.Component<{
         const shareMenuElement = isShareMenuActive && (
             <ShareMenu manager={manager} onDismiss={this.onShareMenu} />
         )
+        const windowPath = getWindowUrl().pathname
+        const relatedQuestionPath = Url.fromURL(
+            relatedQuestions[0]?.url
+        ).pathname
 
         const relatedQuestionElement = relatedQuestions &&
-            hasRelatedQuestion && (
+            hasRelatedQuestion &&
+            // comparing paths rather than full URLs for this to work as
+            // expected on local and staging where the origin (e.g.
+            // hans.owid.cloud) doesn't match the production origin that has
+            // been entered in the related question URL field:
+            // "ourworldindata.org" and yet should still yield a match.
+            // - Note that this won't work on production previews (where the
+            //   path is /admin/posts/preview/ID)
+            windowPath !== relatedQuestionPath && (
```

Comment:
Logic looks good,. Up to you, but I'd lift this up to a variable assignment that can describe it a little bit better, just so that  `relatedQuestionElement` doesn't have all this description in it 👍 

`hasRelatedQuestion && isOnDifferentPath && (`

Reply:
@ikesau makes sense, thanks!